pkg/v15/resource/kubeconfig/desired.go

```diff
@@ -84,7 +84,7 @@ func (r *StateGetter) GetDesiredState(ctx context.Context, obj interface{}) ([]*
 	secret := corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      secretName,
-			Namespace: r.resourceNamespace,
+			Namespace: clusterGuestConfig.ID,
```

Comment:
```suggestion
			Namespace: clusterConfig.ID,
```

Needs renaming here too.